docs/concepts/configuration/pod-priority-preemption.md

```diff
@@ -0,0 +1,177 @@
+numbers are reserved for system use.
+
+PriorityClass also has two optional fields: `globaleDefault` and `description`.
+`globalDefault` indicates that the value of this PriorityClass should be used for
+pods without a `PriorityClassName`. Only one PriorityClass with `globalDefault` 
+set to true can exists in the system. If there is no PriorityClass with `globalDefault`
+set, priority of pods with no `PriorityClassName` will be zero.
+
+`description` is an arbitrary string. It is meant to tell users of the cluster
+when they should use this PriorityClass.
+
+
+**Note 1:** If you upgrade your existing cluster and enable this feature, the priority
+of your existing pods will be considered to be zero.
+
+**Note 2:** Addition of a PriorityClass with `globalDefault` set to true does not
+change priority of existing pods. The value of such PriorityClass will be used only
+for pods created after the PriorityClass is added.
+
+#### Example PriorityClass
+```yaml
+apiVersion: v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+
+## Pod Priority
+Once you have one or more PriorityClasses, you can create pods which specify one
+of those PriorityClass names in their spec.
+The following YAML is an example of a pod configuration that uses the PriorityClass
+created above. Priority admission controller checks the spec and resolves the
+priority of the pod to 1,000,000.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    imagePullPolicy: IfNotPresent
+  priorityClassName: high-priority
+```
+
+## Preemption
+When pods are created, they go to a queue and wait to be scheduled. Scheduler picks a pod
+from the queue and tries to schedule it on a node. If no node is found that satisfies
+all the specified requirements of the pod, the pod is determined infeasible. At this
+point preemption logic is triggered for the pending pod. Let's call the pending pod P.
+Preemption logic tries to find a node where removal of pods with lower priority than
+P helps schedule P. If such node is found, one or more lower priority pods will
+be deleted from the node. Once the pods are gone, P may be scheduled on the node. 
+
+### Limitations of Preemption (alpha version)
+
+#### Starvation of Preempting Pod
+When pods are preempted, the victims get their
+[graceful termination period](https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods).
+They have so much time to finish their work and exit. If they don't, they will be
+killed. This graceful termination period creates a time gap between the point that
+scheduler preempts pods until the pending pod (P) can be scheduled on the node (N).
+When there are multiple victims on node N, they may exit or get terminated at
```

Comment:
Actually, I don't think this problem is reserved to situation where there is many victims - it's just more likely there.

But in general, even if there is one victim, once this is finished you have no guarantee that the first pod you will be processing, will be the preemptor. So someone else may be scheduled in that place.

Reply:
That's correct. I fixed the text.